internal/gensample/gensample.go

```diff
@@ -0,0 +1,737 @@
+			break
+		}
+	}
+	if meth == nil {
+		return errors.E(nil, "service %q doesn't have method %q", serv.GetName(), methConf.Name)
+	}
+
+	inType := g.descInfo.Type[meth.GetInputType()]
+	if inType == nil {
+		return errors.E(nil, "can't find input type %q", meth.GetInputType())
+	}
+	inSpec, err := g.descInfo.ImportSpec(inType)
+	if err != nil {
+		return errors.E(err, "can't import input type: %q", inType)
+	}
+	g.imports[inSpec] = true
+
+	initInfo, err := g.getInitInfo(inType, methConf, sampConf.Request)
+	if err != nil {
+		return err
+	}
+
+	p := g.pt.Printf
+
+	// Region tag
+	argStr, err := argListStr(initInfo, g)
+	if err != nil {
+		return err
+	}
+	p("")
+	p("// [START %s]", sampConf.RegionTag)
+	p("")
+
+	// comments above the sample function
+	requiresNewLine := false
+
+	writeCommentLines := func(comment string) {
+		comment = wrapComment(strings.TrimSpace(comment))
+		writeComment(comment, nil, g)
+	}
+
+	if sampConf.Description != "" {
+		writeCommentLines(fmt.Sprintf("sample%s: %s", meth.GetName(), sampConf.Description))
+		requiresNewLine = true
+	}
+
+	for i, argName := range initInfo.argNames {
+		comment := initInfo.argTrees[i].comment
+		if comment == "" {
+			continue
+		}
+		if requiresNewLine {
+			writeCommentLines("")
+		}
+		requiresNewLine = true
+		writeCommentLines(fmt.Sprintf("%s: %s", argName, comment))
+	}
+
+	// function signature and client initialization
+	p("func sample%s(%s) error {", meth.GetName(), argStr)
+	p("  ctx := context.Background()")
+	p("  c, err := %s.New%sClient(ctx)", g.clientPkg.Name, pbinfo.ReduceServName(serv.GetName(), g.clientPkg.Name))
+	p("  if err != nil {")
+	p("    return err")
+	p("  }")
+	p("")
+	g.imports[pbinfo.ImportSpec{Path: "context"}] = true
+
+	// Set up the request object
+	if err := g.handleRequest(initInfo); err != nil {
+		return err
+	}
+
+	// Make the RPC call and handle output
+	if meth.GetOutputType() == ".google.protobuf.Empty" {
+		err = g.emptyOut(meth, sampConf.Response)
+	} else if meth.GetOutputType() == ".google.longrunning.Operation" {
+		err = g.lro(meth, methConf, sampConf.Response)
+	} else if meth.GetServerStreaming() || meth.GetClientStreaming() {
+		// TODO(hzyi): github.com/googleapis/gapic-generator-go/issues/177
+		err = errors.E(nil, "streaming methods not supported yet")
+	} else if pf, err2 := pagingField(g.descInfo, meth); err2 != nil {
+		err = errors.E(err2, "can't determine whether method is paging")
+	} else if pf != nil {
+		err = g.paging(meth, pf, sampConf.Response)
+	} else {
+		err = g.unary(meth, sampConf.Response)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	p("return nil")
+	p("}")
+	p("")
+	p("// [END %s]", sampConf.RegionTag)
+	p("")
+
+	// main
+	if err := writeMain(g, initInfo.argNames, initInfo.flagNames, initInfo.argTrees, meth.GetName()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *generator) getInitInfo(inType pbinfo.ProtoType, methConf GAPICMethod, fieldConfs []schema_v1p2.RequestConfig) (initInfo, error) {
```

Comment:
function comment: summarize the purpose of this

Reply:
Done.